src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,438 @@
+        const creator: MinimalUser = { id: creatorUser.uid, name: 'creator' };
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        const joiner: Joiner = { ...JoinerMocks.INITIAL, creator };
+        await joinerDAO.set(partId, joiner);
+        await signOut();
+
+
+        const user: FireAuth.User = await createUnverifiedUser('bar@bar.com', 'user');
+        const nonVerifiedUser: MinimalUser = { id: user.uid, name: 'user' };
+        // When adding the non-verified user to the candidates
+        const result: Promise<void> = joinerDAO.addCandidate(partId, nonVerifiedUser);
+
+        // Then it should fail
+        await expectFirebasePermissionDenied(result);
+    });
+    it('should forbid non-creator to change other fields than status', async() => {
+        // Given a part, and a user (that is not the creator)
+        const creatorUser: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const creator: MinimalUser = { id: creatorUser.uid, name: 'creator' };
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+        await signOut();
+
+        const candidateUser: FireAuth.User = await createConnectedGoogleUser('bar@bar.com', 'candidate');
+        const candidate: MinimalUser = { id: candidateUser.uid, name: 'candidate' };
+        // Add ourself to candidate list to try to have more privileges
+        await joinerDAO.addCandidate(partId, candidate);
+
+        const updates: Partial<Joiner>[] = [
+            { chosenOpponent: candidate },
+            { partStatus: PartStatus.CONFIG_PROPOSED.value },
+            { firstPlayer: FirstPlayer.CHOSEN_PLAYER.value },
+            { partType: PartType.BLITZ.value },
+            { maximalMoveDuration: 73 },
+            { totalPartDuration: 1001001 },
+        ];
+        for (const update of updates) {
+            // When modifying a field
+            const result: Promise<void> = joinerDAO.update(partId, update);
+            // Then it should fail
+            await expectFirebasePermissionDenied(result);
+        }
+    });
+    describe('chosen opponent', () => {
+        let partId: string;
+        beforeEach(async() => {
+            // Given a part with a joiner where the current user is set as chosen opponent
+
+            // We have to follow a realistic workflow to achieve that
+            // The part is first created
+            const creatorUser: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'creator');
+            const creator: MinimalUser = { id: creatorUser.uid, name: 'creator' };
+            partId = await partDAO.create(PartMocks.INITIAL);
+            await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+            await signOut();
+
+            // A candidate add itself to the candidates list
```

Comment:
the current user add itself to the candidates list ?
ps: à quel point c'est correct en anglais "it" pour une personne ? ça sonne mieux dans ma tête que "they" en tout cas !

Reply:
pour le ps: "it" c'est pour un objet inanimé, et très insultant si utilisé pour une personne, donc they !

Reply:
et reformulé en neutre correct !

Reply:
J'vois encore des "itself" :p prefer themselves

Reply:
fixed pour themself !

Reply:
Ah bah non, j'en vois encore, y en a sur lesquels tu insistes encore qu'il faut les garder :p ?

Reply:
y'a plus de themself dans ce fichier !